staging/src/k8s.io/apimachinery/pkg/util/net/http.go

```diff
@@ -71,6 +71,8 @@ func IsProbableEOF(err error) bool {
 	switch {
 	case err == io.EOF:
 		return true
+	case err == io.ErrUnexpectedEOF:
+		return true
```

Comment:
The original reason for having the util `IsProbableEOF` was to write fewer errors to logs when watch connections are broken (https://github.com/kubernetes/kubernetes/pull/3037 and https://github.com/kubernetes/kubernetes/pull/5909). I suppose `io.ErrUnexpectedEOF` is not a general connection closure error returned from Go http stack? (@smarterclayton had a good analysis on the go1.3 behavior https://github.com/golang/go/issues/9383#issuecomment-67558070)

According to https://github.com/golang/go/issues/9383 and https://github.com/golang/go/issues/9405, there was a bug in Go when the rest watch client was implemented-- that it was hard to know if an error was caused by timeout. The bug was fixed in go1.5. If we want to refactor/cleanup this code, I'd like to see if we could do better error handling using [net.Error.Timeout()](https://github.com/golang/go/blob/ce58a39fca067a19c505220c0c907ccf32793427/src/net/net.go#L389). Also it would be great if we have integration test to make sure the Go http stack returns error as we expected here.